### ISSUE TITLE
Partially applying a method with an implicit parameter, results in StackOverFlowError

### ISSUE THREAD

**Issue by Eli-Golin** [scala/bug#10544]
```
def foo(implicit a:Int) = a *a   
val moo = foo _
```

The result in REPL:

 ```
java.lang.StackOverflowError
  scala.tools.nsc.typechecker.Contexts$Context.nextOuter(Contexts.scala:615)
  scala.tools.nsc.typechecker.Contexts$Context.implicitss(Contexts.scala:860)
  scala.tools.nsc.typechecker.Contexts$Context.withOuter$1(Contexts.scala:863)
  scala.tools.nsc.typechecker.Contexts$Context.implicitss(Contexts.scala:883)
  scala.tools.nsc.typechecker.Contexts$Context.withOuter$1(Contexts.scala:863)
  scala.tools.nsc.typechecker.Contexts$Context.implicitss(Contexts.scala:883)
  scala.tools.nsc.typechecker.Contexts$Context.withOuter$1(Contexts.scala:864)
  scala.tools.nsc.typechecker.Contexts$Context.implicitss(Contexts.scala:883) 
......
```

Using Scala 2.12.3
Java Oracle 1.8.0_144
OS: Linux Mint 18.2

**Comment by hrhino** [comment]
This seems to be a duplicate of #10332, which is a duplicate of #10279, which duplicates #10299, as well as a few others.

Luckily for you, it's [fixed in 2.12.4](https://github.com/scala/scala/pull/5983).

**Comment by Eli-Golin** [comment]
Tnx
